libc-bottom-half/libpreopen/libpreopen.c

```diff
@@ -494,6 +491,20 @@ __wasilibc_register_preopened_fd(int fd, const char *path)
     return 0;
 }
 
+/// Register the given pre-opened file descriptor under the given path.
+///
+/// This function does not take ownership of `path`.
+int
+__wasilibc_register_preopened_fd(int fd, const char *path)
+{
+    const char *name = strdup(path);
+    if (name == NULL) {
```

Comment:
`return name == NULL ? -1 : __wasilibc_register_preopened_fd(fd, name);` if it convenient with the coding style :).

Reply:
Done!